### PR TITLE
feat: search on directory

### PR DESCRIPTION
**Pull request by 0tuedon**
This PR adds a search functionality while selecting a directory
A Video on How it works below.

https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/1904f7c5-bf0c-4058-b479-42cdb171e6ac

### CODE REVIEW

File: src/components/sideBarContentEdit/SelectDirectory.tsx [hunk 1]
```diff
@@ -102,22 +102,25 @@ const SelectDirectoryOption = ({
...
-            fontSize="14px"
-            px={[2, 4, 6, 9]}
-            py={1}
-          >
-            {dir.value}
-          </Text>
-        ))}
+        options
+          .filter((dir) =>
+            dir.slug.includes(customPath ? customPath.toLowerCase() : "")
```
Comment (BalogunofAfrica) [inline src/components/sideBarContentEdit/SelectDirectory.tsx:107]: Are the items in the slug always guaranteed to be in lowercase?

File: src/components/sideBarContentEdit/SidebarContentEdit.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ const SidebarContentEdit = ({
 
   const updateSpeaker = (speakers: string[]) => {
     const updatedTranscript = getUpdatedTranscript();
-    updatedTranscript.speakers = speakers;
+    updatedTranscript.speakers = speakers;
```
Comment (BalogunofAfrica) [inline src/components/sideBarContentEdit/SidebarContentEdit.tsx:78]: Is the space on this line a mistake?

File: src/components/sideBarContentEdit/SelectDirectory.tsx (hunk 1 above)
Comment (0tuedon) [reply src/components/sideBarContentEdit/SelectDirectory.tsx:107]: yh they are

File: src/components/sideBarContentEdit/SidebarContentEdit.tsx (hunk 2 above)
Comment (0tuedon) [reply src/components/sideBarContentEdit/SidebarContentEdit.tsx:78]: yup

**Pull request merged**